### ISSUE TITLE
Fund account on account creation

### ISSUE THREAD

**Issue by clockworkgr** [bitshares/bitshares-core#1403]
As a `faucet operator` I want `to be able to fund accounts on creation` so that `newly created accounts can contain a balance without having to wait for the account creation operation to be in LIB before performing a transfer`.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by your request. Please tick at least one box.
- [X] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [X] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [X] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [X] UX (the User Experience)
- [ ] Other (please add below)

**Additional Context (optional)**

If this is of interest. I can start drafting a BSIP

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Feature Request
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by clockworkgr** [comment]
Related: https://github.com/bitshares/bitshares-core/issues/1398

**Comment by sschiessl-bcp** [comment]
Many gateways have been asking for this kind of feature, and it would be extremely helpful as well for all the "Insufficient funds" support requests for not properly maintained fee pools. 

The testnet faucet is also supporting that, but via a separate transfer after account creation, which is error prone.

+1

**Comment by xeroc** [comment]
That would be helpful to the testnet faucet too, which gives every new account x TEST for, well, testing ... :)

**Comment by cogutvalera** [comment]
@ryanRfox I want to implement this feature a bit later :) claiming it now if possible, just wanted to mention it ...

Thanks !

**Comment by clockworkgr** [comment]
that was fast :P

**Comment by pmconrad** [comment]
Please write a BSIP.

**Comment by clockworkgr** [comment]
ok. will do

**Comment by ryanRfox** [comment]
Yes, BSIP is the proper next step. @cogutvalera love the enthusiasm. Hope you are willing to assist with drafting the spec.

**Comment by cogutvalera** [comment]
@ryanRfox yes sure thank you !

**Comment by abitmore** [comment]
@clockworkgr asset authorization is not only whitelisting. My bad for didn't state it clearly. IMHO we need to mention it in the BSIP, but don't need to go into details about it.

**Comment by btsabc** [comment]
It looks like we can test the consistency of account_id and name and the consistency of asset_id and symbol by modifying or adding predicate in assert_operation.


https://github.com/bitshares/bitshares-core/blob/3ea52879d634f0bd5bca80e6059b9308ba3270ff/libraries/chain/assert_evaluator.cpp#L33-L71

**Comment by abitmore** [comment]
@btsabc yes a faucet can do that, but also with downsides:
* current fee for assert_operation is expensive;
* the account creation operation or the transfer will be broadcast then fail which is not user-friendly.
